src/lib.rs

```diff
@@ -386,6 +389,33 @@ pub fn fmt_lines(file_map: &mut FileMap, config: &Config) -> FormatReport {
     report
 }
 
+pub fn format_string(input: String, config: &Config, mode: WriteMode) -> FileMap {
+    let path = "stdin";
+    let mut parse_session = ParseSess::new();
+    let krate = parse::parse_crate_from_source_str(path.to_owned(),
+                                                   input,
+                                                   Vec::new(),
+                                                   &parse_session);
+
+    // Suppress error output after parsing.
+    let emitter = Box::new(EmitterWriter::new(Box::new(Vec::new()), None));
+    parse_session.span_diagnostic.handler = Handler::with_emitter(false, emitter);
+
+    // FIXME: we still use a FileMap even though we only have
+    // one file, because fmt_lines requires a FileMap
+    let mut file_map = FileMap::new();
+
+    // do the actual formatting
+    let mut visitor = FmtVisitor::from_codemap(&parse_session, config, Some(mode));
+    visitor.format_separate_mod(&krate.module, path);
+
+    // append final newline
+    visitor.buffer.push_str("\n");
+    file_map.insert(path.to_owned(), visitor.buffer);
+
+    return file_map;
+}
```

Comment:
So after looking at it again, the only code that is actually shared is 
```
    // Suppress error output after parsing.
    let emitter = Box::new(EmitterWriter::new(Box::new(Vec::new()), None));
    parse_session.span_diagnostic.handler = Handler::with_emitter(false, emitter);
```
which isn't worth making a new function IMO. I would much rather try to tackle eliminating the `FileMap` in another PR (as mentioned in the FIXME). Then it will become more apparent that the code in `format_string` is really different from `format`.

So I'd leave it like this for now, if that's ok with you.